notebooks/database.ipynb

```diff
@@ -0,0 +1,854 @@
+       "HBox(children=(FloatProgress(value=1.0, bar_style='info', description='Saving to database', layout=Layout(widt…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "database.save(runs_low)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>climate_model</th>\n",
+       "      <th>scenario</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>model_a</td>\n",
+       "      <td>low</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>model_b</td>\n",
+       "      <td>low</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "  climate_model scenario\n",
+       "0       model_a      low\n",
+       "1       model_b      low"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "database.available_data()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Internally, these each row shown in `available_data()` is stored as a netCDF file in a directory structure following ``database.levels``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[01;34m.\u001b[00m\r\n",
+      "├── \u001b[01;34mmodel_a\u001b[00m\r\n",
+      "│   └── \u001b[01;34mlow\u001b[00m\r\n",
+      "│       └── model_a_low.nc\r\n",
+      "└── \u001b[01;34mmodel_b\u001b[00m\r\n",
+      "    └── \u001b[01;34mlow\u001b[00m\r\n",
+      "        └── model_b_low.nc\r\n",
+      "\r\n",
+      "4 directories, 2 files\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pushd {temp_out_dir.name}; tree; popd"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Additional calls to `save` will merge the new data into the database, creating any new files as required.\n",
+    "\n",
+    "If existing data is found, it is first loaded and merged with the saved data before writing so not to lose existing data. "
```

Comment:
```suggestion
    "If existing data is found, it is first loaded and merged with the saved data before writing to prevent losing existing data. "
```